src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -999,3 +1004,54 @@ winrt::hstring CascadiaSettings::ApplicationVersion()
 
     return RS_(L"ApplicationVersionUnknown");
 }
+
+// Method Description:
+// - Forces a refresh of all default terminal state
+// Arguments:
+// - <none>
+// Return Value:
+// - <none> - Updates internal state
+void CascadiaSettings::RefreshDefaultTerminals()
+{
+    _defaultTerminals.Clear();
+
+    for (const auto& term : Model::DefaultTerminal::Available())
+    {
+        _defaultTerminals.Append(term);
+    }
+
+    _currentDefaultTerminal = Model::DefaultTerminal::Current();
+}
+
+// Method Description:
+// - Returns an iterable collection of all available terminals.
+// Arguments:
+// - <none>
+// Return Value:
+// - an iterable collection of all available terminals that could be the default.
+IObservableVector<winrt::Microsoft::Terminal::Settings::Model::DefaultTerminal> CascadiaSettings::DefaultTerminals() const noexcept
+{
+    return _defaultTerminals;

```

Comment:
Should we check if `_defaultTerminals` is empty? And if it is, we call `RefreshDefaultTerminals`?

Reply:
Breaks const. Can't do it.